security/pkg/nodeagent/sds/server.go

```diff
@@ -41,21 +38,37 @@ type Server struct {
 	grpcWorkloadListener net.Listener
 
 	grpcWorkloadServer *grpc.Server
-	debugServer        *http.Server
 }
 
 // NewServer creates and starts the Grpc server for SDS.
-func NewServer(options *ca2.Options, workloadSecretCache ca2.SecretManager) (*Server, error) {
-	s := &Server{
-		workloadSds: newSDSService(workloadSecretCache, options, options.FileMountedCerts),
+func NewServer(options security.Options, workloadSecretCache security.SecretManager) (*Server, error) {
+	s := &Server{}
+	if err := FillServer(s, options, workloadSecretCache); err != nil {
+		return nil, err
 	}
+	return s, nil
+}
+
+func FillServer(s *Server, options security.Options, workloadSecretCache security.SecretManager) error {
```

Comment:
Comment above FillServer?

Reply:
I removed the function, forgot to clean it up before